rmgpy/cantherm/molepro.py

```diff
@@ -42,7 +45,196 @@ class MoleProLog:
     
     def __init__(self, path):
         self.path = path
-            
+
+    def getNumberOfAtoms(self):
+        """
+        Return the number of atoms in the molecular configuration used in
+        the MolPro log file.
+        """
+
+        Natoms = 0
+        # Open Molpro log file for parsing
+        f = open(self.path, 'r')
+        line = f.readline()
+        while line != '' and Natoms == 0:
+            # Automatically determine the number of atoms
+            if 'ATOMIC COORDINATES' in line and Natoms == 0:
+                for i in range(4): line = f.readline()
+                while 'Bond lengths' not in line:
+                    Natoms += 1
+                    line = f.readline()
+            line = f.readline()
+        # Close file when finished
+        f.close()
+        # Return the result
+
+        return Natoms - 1
+
+    def loadForceConstantMatrix(self):
+        """
+        Print the force constant matrix by including the print, hessian command in the input file
+        """
+
+        F = None
+
+        Natoms = self.getNumberOfAtoms()
+        Nrows = Natoms * 3
+
+        f = open(self.path, 'r')
+        line = f.readline()
+        while line != '':
+            # Read force constant matrix
+            if 'Force Constants (Second Derivatives of the Energy) in [a.u.]' in line:
+                F = numpy.zeros((Nrows,Nrows), numpy.float64)
+                for i in range(int(math.ceil(Nrows / 5.0))):
+                    # Header row
+                    line = f.readline()
+                    # Matrix element rows
+                    for j in range(i*5, Nrows):
+                        data = f.readline().split()
+                        for k in range(len(data)-1):
+                            F[j,i*5+k] = float(data[k+1].replace('D', 'E'))
+                            F[i*5+k,j] = F[j,i*5+k]
+                # Convert from atomic units (Hartree/Bohr_radius^2) to J/m^2
+                F *= 4.35974417e-18 / 5.291772108e-11**2
+            line = f.readline()
+        # Close file when finished
+        f.close()
+
+        return F
+
+    def loadGeometry(self):
+        """
+        Return the optimum geometry of the molecular configuration from the
+        Molpro .out file. If multiple such geometries are identified, only the
+        last is returned.
+        """
+
+        symbol = []; coord = []
+
+        f = open(self.path, 'r')
+        line = f.readline()
+        while line != '':
+            # Automatically determine the number of atoms
+            if 'Current geometry' in line:
+                symbol = []; coord = []
+                for i in range(4): line = f.readline()
+                while line != '\n':
```

Comment:
I have a slightly different syntax for the molpro opt:
```

 END OF GEOMETRY OPTIMIZATION.

 Current geometry (xyz format, in Angstrom)


 Dump information in style XYZ to geometry.xyz                                                                    

    5
 DF-RKS-SCF000/CC-PVTZ  ENERGY=-698.91289988
 S         -0.0000000000       -0.0000000000        0.2325737348
 O         -0.0000000000        0.7683688596       -1.1496180790
 O         -1.2455119069        0.0000000000        0.9137857116
 O         -0.0000000000       -0.7683688596       -1.1496180790
 O          1.2455119069        0.0000000000        0.9137857116
```
Counting 4 lines from the ` Current geometry` phrase leads to an empty line in this case, and the geometry isn't parsed. We need a more robust method here. Maybe search for the word `ENERGY` after the `Current geometry` phrase, the parse the geometry?